src/components/Booklist.js

```diff
@@ -1,24 +1,31 @@
 import { useSelector } from 'react-redux';
 import Book from './Book';
-// import booksSlice from '../redux/books/booksSlice';
 
 function BookList() {
   const {
     booksList,
   } = useSelector((state) => state.books);
 
+  if (!Array.isArray(booksList)) {
+    console.log('booklist is not an array');
+    return null; // or return an appropriate JSX if the data is not available yet
+  }
+
   return (
     <>
       {booksList.map((item) => (
-        <Book
-          key={item.id}
-          id={item.id}
-          category={item.category}
-          title={item.title}
-          author={item.author}
-          percentage={item.percentage}
-          chapter={item.chapter}
-        />
+        // console.log()
+        <>
```

Comment:
- [ ] Please, let's try not to have commented unused code in our codebase as this makes our code not neat and not professional. It's a good practice to not have commented unused codes in our codebase. Kindly remove it 👍🏽